### PR TITLE
Add s3 adapter

### PR DESCRIPTION
**Pull request by kmcrawford**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,31 +1,40 @@
...
+      "git add"
+    ]
+  },
+  "jest": {
+    "rootDir": "packages",
+    "coverageDirectory": "./coverage/",
+    "collectCoverage": true
+  },
+  "dependencies": {
+    "async": "^2.5.0",
```
Comment (nickbalestra) [inline package.json]: Long: One rule we try to follow is that the root package.json should be dependencies free, and only have devDependencies), the opposite apply for inner packages: they should be devDependencies free as their devDependencies could be simply moved to the devDependencies of the root manifest.
In this way the root work as the orchestrator for everything needed for dev and publishing, while each package is simply let to handle only its own dependencies. A nice thing with this is that you have consistency devDependencies across the whole monorepo.

TLDR: don't put those dependencies here, but only on the packages that needs them

File: packages/oc-s3-storage-adapter/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "url": "https://github.com/opencomponents/storage-adapters/issues"
+  },
+  "homepage": "https://github.com/opencomponents/storage-adapters#readme",
+  "keywords": [
+    "oc",
+    "opencomponents",
+    "s3"
+  ],
+  "author": {
+    "name": "Nick Balestra",
```
Comment (nickbalestra) [inline packages/oc-s3-storage-adapter/package.json]: I guess you want to put your name here :)

**Comment by nickbalestra** [comment]
Awesome @kmcrawford tomorrow I'll review and if all ok we can merge and sync up for publishing this, and in parallel work on a pr to OC so that we can consume this. Great work, thanks alot for helping with this!

File: .gitignore [hunk 3]
```diff
@@ -22,4 +22,5 @@ results
 npm-debug.log
 node_modules
 coverage
-/mocks/handlebars-component/_compile-static-package6/
\ No newline at end of file
+/mocks/handlebars-component/_compile-static-package6/
+package-lock.json
```
Comment (nickbalestra) [inline .gitignore]: I see that a `yarn.lock` made it through (packages/oc-s3-storage-adapter/yarn.lock), as we haven't yet started adding locks file in any of our repo, perhaps we can also remove the yarn lock file and ignore it as well here?

File: package.json [hunk 4]
```diff
@@ -1,31 +1,38 @@
...
+    "collectCoverage": true
+  },
+  "devDependencies": {
+    "codecov": "^2.2.0",
+    "husky": "^0.14.3",
+    "jest": "^21.2.1",
+    "lerna": "^2.3.1",
+    "lint-staged": "^4.2.3",
+    "prettier-eslint-cli": "^4.4.0",
+    "async": "^2.5.0",
```
Comment (nickbalestra) [inline package.json]: I think there is no need anymore for those* here they are now correctly defined as dependencies in the respective packages (ie: `packages/oc-s3-storage-adapter/package.json`).


- async
- aws-sdk
- fs-extra
- lodash
- nice-cache
- node-dir
- stringformat
